### DISCUSSION TITLE
Creation of DFSET from CRYSTAL17 output

### DISCUSSION THREAD

**Discussion by RicardoCecconello** [ttadano/alamode#202] [Ideas]
Dear Dr. Tadano,

We will do some calculations with CRYSTAL17 and would like to collect the displacements and forces data into a DFSET file for usage with ALAMODE. As far as we know, to calculate the forces we need to set a geometry optimization, from which we can extract the forces of the input structure from the first output parameters. I am sending you the OUTPUT and the FORCES.DAT files created. In the latter, the first block has the cartesian coordinates and the forces in hartree/bohr.

Also, I was wondering if it is possible for you to supply a code for the extraction. If so, I would be very grateful!

[OUTPUT.txt](https://github.com/user-attachments/files/15568987/OUTPUT.txt)
[FORCES.txt](https://github.com/user-attachments/files/15568988/FORCES.txt)

**Comment by ttadano** [comment]
Hi,

Converting `FORCES.txt` into the DFSET format of ALAMODE would be easy. What you need to do is just convert the coordinates into displacements from a relaxed structure, and change the units of displacements and forces into bohr and Ry/bohr, respectively.

Unfortunately, I cannot write a parser for the CRYSTAL code because I do not have a license of it. I hope someone else can contribute to this.

**Comment by RicardoCecconello** [comment]
Alright, thank you!
